Make working progress bar

shaavan opened the pull request:
- This PR makes the progress bar on the block sync page work.
- This was done by introducing a new private member in the NodeModel class.

**Screenshots:**

**Master**

![Screenshot from 2021-12-28 11-38-51](https://user-images.githubusercontent.com/85434418/147533861-a040157a-7d5a-48b8-a3ce-452d98108613.png)

**PR (at different instances)**

|60680 Blocks|65344 Blocks|69744 Blocks|
|----|----|----|
|![Screenshot from 2021-12-28 11-35-27](https://user-images.githubusercontent.com/85434418/147533879-fb7693be-3bce-43bf-9b85-0f8008a6291d.png)|![Screenshot from 2021-12-28 11-35-34](https://user-images.githubusercontent.com/85434418/147533893-6c99a97d-a9c0-44bb-bf8a-d629949fb18b.png)|![Screenshot from 2021-12-28 11-35-41](https://user-images.githubusercontent.com/85434418/147533914-53718f20-112a-43ed-aae4-9306158a39b1.png)|

shaavan commented:
Changes in following files:
- gui-qml.cflags
- gui-qml.config
- gui.qml.creator
- gui-qml.cxxflags
- gui-qml.files
- gui-qml.includes

They are created automatically by the Qt Creator.

If any of these changes are redundant or unwanted, kindly inform me, and I shall remove them in the next update of the PR.
Thank You.

shaavan commented:
To test the PR, you can use the following command:

```
./src/qt/bitcoin-qt -signet -daemon -reindex
```

jarolrod commented:
Please remove all of the Qt Creator specific files.

For now this is not exactly needed. You can leave this open for now, I'll include this as part of my onboarding PR with some changes.

shaavan commented:
Updated from fa36f92b80ff1b8359bdc4a49f280595e31c1da0 to ff40e069ef1947a7354afafa6d49dceb6ebeb5f5 ([pr99.01](https://github.com/shaavan/gui-qml/tree/pr99.01) -> [pr99.02](https://github.com/shaavan/gui-qml/tree/pr99.02))
Addressed @jarolrod comments

Changes:

- Removed Qt creator-specific changes from the PR.

> I'll include this as part of my onboarding PR with some changes.

Sound great!

promag commented:
ACK exposing the verification progress property so it can be binded elsewhere.

hebasto commented on src/qml/nodemodel.cpp:
```diff
@@ -23,6 +23,14 @@ void NodeModel::setBlockTipHeight(int new_height)
 
+void NodeModel::setVerificationProgress(double new_progress)
+{
+    if(new_progress != m_verification_progress) {
```
style:
```suggestion
    if (new_progress != m_verification_progress) {
```

hebasto commented on src/qml/nodemodel.cpp:
```diff
@@ -37,13 +45,17 @@ void NodeModel::initializeResult([[maybe_unused]] bool success, interfaces::Bloc
 void NodeModel::ConnectToBlockTipSignal()
 {
     assert(!m_handler_notify_block_tip);
+    assert(!m_verification_progress);
+
```
Why this assertion?

shaavan replied:
That should not have been there. I was doing some trying and testing but forgot to remove this statement when I was done.
Thanks for catching this.
Fixed in da06542c7968077d544906ef3e6fbfc56416dcca

hebasto reviewed (commented):
Concept ACK.

Could you please split changes in `src/qml/pages/stub.qml` into a separated commit? Expecting it will improve maintainability (e.g., during syncing with the main repo). For example, see b6280a9c022ae5ac0c5c093ab77d86fd5bbbc652 and 1441d5be79f161e7864bf5493b5ed68c51731eec.

Also it'd nice if QML-only-stuff commits have a `qml:` prefix.

shaavan commented:
Updated from ff40e069ef1947a7354afafa6d49dceb6ebeb5f5 to da06542c7968077d544906ef3e6fbfc56416dcca ([pr99.02](https://github.com/shaavan/gui-qml/tree/pr99.02) -> [pr99.03](https://github.com/shaavan/gui-qml/tree/pr99.03), [diff](https://github.com/shaavan/gui-qml/compare/pr99.02..pr99.03))
Addressed @hebasto comments.

Changes:

- Fixed styling
- Remove unnecessary assert.

hebasto commented on src/qml/pages/stub.qml:
```diff
@@ -35,7 +35,7 @@ ApplicationWindow {
             id: indicator
             Layout.fillWidth: true
-            progress: 0.666
+            progress: nodeModel.verificationProgress
```
Should we drop the following `background: MouseArea` now?

hebasto reviewed (commented):
> Also it'd nice if QML-only-stuff commits have a `qml:` prefix.

Please add "qml: " to the beginning of your commit message.

hebasto commented:
@shaavan 

And please split in two commits (https://github.com/bitcoin-core/gui-qml/pull/99#pullrequestreview-844475025).

shaavan commented:
Updated from da06542c7968077d544906ef3e6fbfc56416dcca to bcea86c33ed98dfe11c6c7a8153032350c9165f8 ([pr99.03](https://github.com/shaavan/gui-qml/tree/pr99.03) -> [pr99.04](https://github.com/shaavan/gui-qml/tree/pr99.04), [diff](https://github.com/shaavan/gui-qml/compare/pr99.03..pr99.04))
Addressed @hebasto comments.

Changes:

- Split PR into two commits. Added `qml:` prefix to each of them
- Removed temporary mouse click event from the progress bar.

hebasto reviewed (approved):
ACK bcea86c33ed98dfe11c6c7a8153032350c9165f8

promag reviewed (commented):
ACK bcea86c33ed98dfe11c6c7a8153032350c9165f8
